Solitaire/js/main.js

```diff
@@ -1,4 +1,259 @@
+		deckFinish3: {array: [], click: checkCard},
+		deckFinish4: {array: [], click: checkCard}
+	}
+	var mixRandom = function() {
+        return Math.random() - 0.5;
+    };
+	var deckSelected = null;
+    var cardSelectedNumber = null;
+
+    
+    function fillingArrayCards() {
+
+        for (var i = 1; i < 53; i++) {
+            var value = i%13 === 0 ? 13 : i%13;
+            var name;
+            var suit;
+            var suitGroup = Math.ceil(i/13);
+    
+            if (suitGroup === 1) {
+                suit = 'hearts';
+            } else if (suitGroup === 2) {
+                suit = 'diamonds';
+            } else if (suitGroup === 3) {
+                suit = 'clovers';
+            } else if (suitGroup === 4) {
+                suit = 'spades';
+            }
+    
+            if (value === 1) {
+                name = 'A';
+            } else if (value === 11) {
+                name = 'J';
+            } else if (value === 12) {
+                name = 'Q';
+            } else if (value === 13) {
+                name = 'K';
+            } else {
+                name = String(value);
+            }
+    
+            cards.push({
+                isVisible: false,
+                id: 'card' + i,
+                value: value,
+                name: name,
+                suit: suit,
+            });
+        }
+    };
+
+    function fillingGameDecks() {
+		for (var i = 0; i < cards.length; i++) {
+			for (key in playingDecks) {
+				if (playingDecks[key].cardPos && i < playingDecks[key].cardPos) {
+					playingDecks[key].array.push(cards[i]);
+					
+					break;
+				}
+				
+			}
+		}
+		
+		for (key in playingDecks) {
+			if (playingDecks[key].cardPos) {
+				console.log(key);
+				if (playingDecks[key] === playingDecks.deckDistribution) {
+					playingDecks[key].array[playingDecks[key].array.length-1].isVisible = false;
+				}else {
+					playingDecks[key].array[playingDecks[key].array.length-1].isVisible = true;
+				}
+			}
+		}
+    };
+  
+    function drawTable(){
+		for (key in playingDecks) {
+			createDeck(playingDecks[key].array, document.getElementById(key));
+		}
+    };
+
+    function createDeck(deck, $deck){
+        var cardsHTML = '';
+
+        for (var i=0;i<deck.length;i++) {
+            cardsHTML = cardsHTML + '<div id="'+ deck[i].id +'" class="card '+ deck[i].suit + (deck[i].isVisible ? ' open' : '') +'">'+ deck[i].name +'</div>';
+        }
+        $deck.innerHTML = cardsHTML;
+    };
+       
+    function addEvents(){
+		for (key in playingDecks) {
+			(function(deckKey){
+				document.getElementById(deckKey).addEventListener('click', function(event) {
+					playingDecks[deckKey].click(event, deckKey);
+				});
+			})(key);
+		}
+
+        $btnReset.addEventListener("click", function () {
+            resetGame();
+        });
+
+
+    };
+
+    function openCardDestribution() {
+        var clickedCard;
+        var card;
+
+        if (!playingDecks.deckDistribution.array.length) {
```

Comment:
don't use negative comparisons if it's possible. in this case you can remove `!` and exchange the **if** and **else** bodies